### ISSUE TITLE
Allow nested dictionaries in IPC

### ISSUE THREAD

**Issue by jorgecarleitao** [jorgecarleitao/arrow2#499]
Basically, add a new `generated_nested_dictionaries` test file,

```rust
#[test]
fn read_generated_nested_dictionary() -> Result<()> {
    test_file("1.0.0-littleendian", "generated_nested_dictionary")?;
    test_file("1.0.0-bigendian", "generated_nested_dictionary")
}
```

and make it pass without breaking other tests.

There is also a comment

```rust
// TODO: handle nested dictionaries
```

**Comment by xudong963** [comment]
I'll do it @jorgecarleitao

**Comment by jorgecarleitao** [comment]
This has been closed by #587
